nixos/tests/nvidia-container-toolkit.nix

```diff
@@ -0,0 +1,158 @@
+import ./make-test-python.nix (
+  {
+    pkgs,
+    lib,
+    system,
+    ...
+  }:
+  let
+    unfreeAndInsecurePkgs = import ../.. {
+      inherit system;
+      config = {
+        allowUnfree = true;
+        permittedInsecurePackages = [ "openssl-1.1.1w" ];
+      };
+    };
+    testContainerImage =
+      let
+        testCDIScript = pkgs.writeShellScriptBin "test-cdi" ''
+          die() {
+            echo "$1"
+            exit 1
+          }
+
+          check_file_referential_integrity() {
+            echo "checking $file referential integrity"
+            files=$(set -o pipefail && \
+                    ${pkgs.glibc.bin}/bin/ldd "$1" | \
+                    ${pkgs.gnugrep}/bin/grep '=>' | \
+                    ${pkgs.gnused}/bin/sed "s/.* => //" | \
+                    ${pkgs.gnused}/bin/sed "s/ (.*//") || exit 1
+
+            for file in $files; do
+              if [ ! -f "$file" ]; then
+                die "$file does not exist in the container filesystem"
```

Comment:
I think if a path was reported on the RHS of `=>` it means it was loaded, which means it does exist

Reply:
Right, I guess I can grep for "not found" then. I don't love this poor man referential integrity check TBH.

Reply:
Aah I see, we don't automatically know if a line is a path or a not-found 👍🏻